src/components/marker-chart/Canvas.js

```diff
@@ -192,8 +194,21 @@ class MarkerChartCanvas extends React.PureComponent<Props, State> {
           markerTiming.end[i] > timeAtViewportLeft &&
           markerTiming.start[i] < timeAtViewportRightPlusMargin
         ) {
+          let markerTimingStart = markerTiming.start[i];
+          if (markerTimingStart < 0 && thread.samples.time[0] !== undefined) {
+            // If the marker is a tracing marker with only the end
+            // part (indicated by negative start), start the drawing
+            // of the marker from the time where the first sample was
+            // taken. It can happen that this occurs _after_ the
+            // actual end marker. In that case, we start drawing from
+            // the end instead and we'll get a duration of 0.
+            markerTimingStart = Math.min(
+              thread.samples.time[0],
+              markerTiming.end[i]
+            );
+          }
```

Comment:
Actually I believe that the change should happen in this code instead:
https://github.com/devtools-html/perf.html/blob/68c36b3ef032598afc969f436da2e53108d66316/src/profile-logic/marker-data.js#L281-L287